store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java

```diff
@@ -0,0 +1,526 @@
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.volume;
+
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.OptionGroup;
+import org.apache.commons.cli.Options;
+
+import com.zimbra.cs.service.admin.AdminService;
+import com.zimbra.cs.util.BuildInfo;
+import com.zimbra.cs.util.SoapCLI;
+import com.zimbra.common.auth.ZAuthToken;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.soap.GlobalExternalStoreConfigConstants;
+import com.zimbra.common.soap.SoapTransport;
+import com.zimbra.common.util.CliUtil;
+import com.zimbra.common.util.StringUtil;
+import com.zimbra.soap.admin.type.Attr;
+import com.zimbra.soap.admin.message.CreateS3BucketConfigRequest;
+import com.zimbra.soap.admin.message.CreateS3BucketConfigResponse;
+import com.zimbra.soap.admin.message.GetS3BucketConfigRequest;
+import com.zimbra.soap.admin.message.GetS3BucketConfigResponse;
+import com.zimbra.soap.admin.message.DeleteS3BucketConfigRequest;
+import com.zimbra.soap.admin.message.DeleteS3BucketConfigResponse;
+import com.zimbra.soap.type.GlobalS3BucketConfiguration;
+import com.zimbra.soap.JaxbUtil;
+
+public final class VolumeS3ConfigCLI extends SoapCLI {
+    // cli constants
+    private static final String  O_A    = "a";
+    private static final String  O_D    = "d";
+    private static final String  O_L    = "l";
+    private static final String  O_BID  = "bid";
+    private static final String  O_BN   = "bn";
+    private static final String  O_AK   = "ak";
+    private static final String  O_SK   = "sk";
+    private static final String  O_DP   = "dp";
+    private static final String  O_R    = "r";
+    private static final String  O_ACC  = "acc";
+    private static final String  O_NS   = "ns";
+    private static final String  O_PP   = "pp";
+    private static final String  O_URL  = "url";
+    private static final String  O_SP   = "t";
+    // store provider constants
+    private static final String AWS_S3 = "AWS_S3";
+    private static final String CEPH_S3 = "CEPH_S3";
+    private static final String STORAGE_GRID_S3 = "CEPH_S3";
+    // variables
+    private ZAuthToken auth;
+    private String storeProvider = "";
+    private String bucketId = "";
+    private String bucketName = "";
+    private String accessKey = "";;
+    private String secretKey = "";
+    private String destPath = "";
+    private String region = "";
+    private String account = "";
```

Comment:
You don't need following parameters for S3 
 private String account = "";
  private String namespace = "";
    private String proxyPort = "";
    private String accountPort = "";
    private String url = "";

Reply:
Removed unnecessary params